core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -587,10 +595,50 @@ public void callOptionsExecutor() {
   @Test
   public void nameResolutionFailed() {
     Status error = Status.UNAVAILABLE.withCause(new Throwable("fake name resolution error"));
+    FakeNameResolverFactory nameResolverFactory = new FakeNameResolverFactory(error);
+    // Name resolution is started as soon as channel is created.
+    createChannel(nameResolverFactory, NO_INTERCEPTOR);
+    FakeNameResolverFactory.FakeNameResolver resolver = nameResolverFactory.resolvers.get(0);
+    verify(mockLoadBalancer).handleNameResolutionError(same(error));
 
+    timer.forwardNanos(RECONNECT_BACKOFF_INTERVAL_NANOS);
```

Comment:
To verify refresh() is not called sooner than expected, you will need to forward by `RECONNECT_BACKOFF_INTERVAL_NANOS - 1` and verify that refresh() is not called, then forward by 1.

Reply:
Done